app/liquid_tags/youtube_tag.rb

```diff
@@ -32,42 +35,41 @@ def render(_context)
   private
 
   def parse_id_or_url(input)
-    if (input =~ REGISTRY_REGEXP)&.zero?
-      extract_youtube_id(input)
-    else
-      input_no_space = input.delete(" ")
-      raise StandardError, "Invalid YouTube ID" unless valid_id?(input_no_space)
-      return translate_start_time(input_no_space) if input_no_space.include?("?t=")
-
-      input_no_space
-    end
-  end
+    match = pattern_match_for(input)
+    raise StandardError, "Invalid YouTube ID" unless match
+
+    video_id       = match[:video_id]
+    time_parameter = match[:time_parameter]
 
-  def extract_youtube_id(url)
-    url = url.gsub(/(>|<)/, "").split(%r{(vi/|v=|/v/|youtu\.be/|/embed/)})
-    raise StandardError, "Invalid YouTube URL" if url[2].nil?
+    return video_id if time_parameter.blank?
 
-    id = url[2].split(/[^a-zA-Z0-9_-]/i) # tweak this to allow for time, fix youtube_tag_spec
-    id[0]
+    translate_start_time(video_id, time_parameter)
   end
 
-  def valid_id?(id)
-    id.match?(/\A[a-zA-Z0-9_-]{11}((\?t=)?(\d{1,}h?)?(\d{1,2}m)?(\d{1,2}s)?){5,11}?\Z/)
+  def pattern_match_for(input)
+    match = nil
+    REGEXP_OPTIONS.each do |regex|
+      m = input.match(regex)
+      if m
+        match = m
+        break
+      end
+    end
+    match
```

Comment:
This seems like it does the accomplishes the same goal: it finds the first element in the array where the block is truthy and returns the value of that block.

```suggestion
    REGEXP_OPTIONS
      .filter_map { |regex| input.match(regex) }
      .first
```

I don't know why `Enumerable` doesn't have a short-circuiting method for this (like how `find` short-circuits when it hits the first truthy result so we don't have to use `.first` and avoids creating an intermediate array) but this is a pretty reasonable compromise. The short-circuiting approach is a bit more verbose.

Reply:
The short-circuit approach, for the sake of completeness:

```ruby
REGEXP_OPTIONS.find do |regex|
  if (match = input.match(regex))
    break match
  end
end
```